power_grid_model_io_native_c/power_grid_model_io_native/CMakeLists.txt

```diff
@@ -2,8 +2,10 @@
 #
 # SPDX-License-Identifier: MPL-2.0
 
-add_library(power_grid_model_io_native INTERFACE)
+add_library(power_grid_model_io_native STATIC) #static
```

Comment:
please reorganize the header files such that you have a single `STATIC` or `INTERFACE` library and then another separate `SHARED` library

right now, the static and the shared library share headers and sources and in that case, you might as well only have the shared library